### ISSUE TITLE
Republish ahash

### ISSUE THREAD

**Issue by BartMassey** [tats-u/rust-oem-cp#3]
Since `ahash::AHashMap` is part of the public interface of this crate, it would be useful to republish `ahash`. This would allow the use of `AHashMap` from this crate by crate client. PR #2 would close this issue.

**Comment by tats-u** [comment]
Could you tell me what type of error occurs without republishing `ahash`?
I have just decided to wrap `AHashMap` to make the API consistent (e.g. when changing the hashmap to something else) in the next ~~major~~ version ~~(v1.0)~~.

**Comment by BartMassey** [comment]
Wrapping `AHashMap` would be great.

Consider this code:

```rust
   let encode: &'static ahash::AHashMap<char, u8> =
        *(*oem_cp::code_table::ENCODING_TABLE_CP_MAP).get(&0).unwrap();
```

won't work because

```
error[E0308]: mismatched types                                                  
 --> src/lib.rs:3:9
  |
2 |     let encode: &'static ahash::AHashMap<char, u8> =
  |                 ---------------------------------- expected due to this
3 |         *(*oem_cp::code_table::ENCODING_TABLE_CP_MAP).get(&0).unwrap();
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expe\
cted struct `AHashMap`, found struct `ahash::hash_map::AHashMap`
  |
  = note: expected reference `&'static AHashMap<char, u8>`
             found reference `&ahash::hash_map::AHashMap<char, u8>`
  = note: perhaps two different versions of crate `ahash` are being used?
```

If I instead try `oem_cp::AHashMap` I get a privacy violation as expected. I could probably add a `patch` to my `Cargo.toml` to work around this, but that seems fragile.

**Comment by tats-u** [comment]
@BartMassey 
Ah, this would be troublesome.
I propose a different way to fix your problem.  Do you think #4 is a sufficient solution?
I may change the algorithm of the hashmap in this crate in the future, so it is not good to make the specific algorithm name public.
This change is not so much breaking one because [my program using this crate](https://github.com/tats-u/zifu) does not specify the type name `AHashMap` directly.
This change will bumps the version up to 1.0.

**Comment by BartMassey** [comment]
I think that will be great for now! I don't see the version number change to `Cargo.toml` in the patch, though. :-)

Again, thanks much for your quick response, and for a really nice crate.

**Comment by BartMassey** [comment]
Thanks much!

**Comment by tats-u** [comment]
Not at all.
I have just [released v1.0.0](https://github.com/tats-u/rust-oem-cp/releases/tag/v1.0.0).
Thank you for pointing out the defect of the API, too.
